### PR TITLE
add support for single arrows

### PR DESCRIPTION
**Pull request by meszaros-lajos-gyorgy**

### CODE REVIEW

File: src/core/Core.cpp [hunk 1]
```diff
@@ -752,10 +752,20 @@ void ManageCombatModeAnimations() {
 			if(layer1.cur_anim == alist[ANIM_MISSILE_WAIT]) {
 				player.m_aimTime = PlatformDuration::ofRaw(1);
 				
-				if(eeMousePressed1() && getInventoryItemWithLowestDurability("arrows", 1.f) != nullptr) {
-					changeAnimation(io, 1, alist[ANIM_MISSILE_STRIKE_PART_1]);
-					io->isHit = false;
+				if(eeMousePressed1()) {
+					if (getInventoryItemWithLowestDurability("arrow", 1.f) != nullptr) {
```
Comment (fredlllll) [inline src/core/Core.cpp:756]: you could put these 2 conditions into an OR as the bodies of the blocks seem to be the same?

File: src/core/Core.cpp [hunk 2]
```diff
@@ -752,10 +752,20 @@ void ManageCombatModeAnimations() {
...
+						changeAnimation(io, 1, alist[ANIM_MISSILE_STRIKE_PART_1]);
+						io->isHit = false;
+					} else if (getInventoryItemWithLowestDurability("arrows", 1.f) != nullptr) {
+						changeAnimation(io, 1, alist[ANIM_MISSILE_STRIKE_PART_1]);
+						io->isHit = false;
+					}
 				}
+
+				// if(eeMousePressed1() && getInventoryItemWithLowestDurability("arrows", 1.f) != nullptr) {
```
Comment (fredlllll) [inline src/core/Core.cpp:765]: the point of source control is that you dont have to keep comments of old code around :P

File: src/core/Core.cpp [hunk 3]
```diff
@@ -849,29 +859,36 @@ void ManageCombatModeAnimations() {
...
 				EERIE_LINKEDOBJ_UnLinkObjectFromObject(io->obj, arrowobj.get());
 				changeAnimation(io, 1, alist[ANIM_MISSILE_STRIKE]);
 				SendIOScriptEvent(nullptr, io, SM_STRIKE, "bow");
 				StrikeAimtime();
 				player.m_strikeAimRatio = player.m_bowAimRatio;
-				Entity * quiver = getInventoryItemWithLowestDurability("arrows", 1.f);
+
 				float poisonous = 0.f;
+				Entity * arrow = getInventoryItemWithLowestDurability("arrow", 1.f);
+				if (arrow) {
```
Comment (fredlllll) [inline src/core/Core.cpp:870]: why do a comparison to nullptr further up but not here?

File: src/core/Core.cpp [hunk 4]
```diff
@@ -849,29 +859,36 @@ void ManageCombatModeAnimations() {
...
+
+					poisonous = arrow->poisonous;
+				} else {
+					Entity * quiver = getInventoryItemWithLowestDurability("arrows", 1.f);
 				
-				if(quiver) {
-					poisonous = quiver->poisonous;
-					if(quiver->poisonous_count > 0) {
-						quiver->poisonous_count--;
+					if(quiver) {
```
Comment (fredlllll) [inline src/core/Core.cpp:877]: also no comparison to nullptr, either dont do it at all, or do it everywhere, consistency is key :P

**Review by fredlllll — CHANGES_REQUESTED** [review]

File: src/core/Core.cpp [hunk 5]
```diff
@@ -752,7 +752,7 @@ void ManageCombatModeAnimations() {
 			if(layer1.cur_anim == alist[ANIM_MISSILE_WAIT]) {
 				player.m_aimTime = PlatformDuration::ofRaw(1);
 				
-				if(eeMousePressed1() && getInventoryItemWithLowestDurability("arrows", 1.f) != nullptr) {
+				if(eeMousePressed1() && getInventoryItemWithLowestDurability("arrow", 1.f) != nullptr && getInventoryItemWithLowestDurability("arrows", 1.f) != nullptr) {
```
Comment (fredlllll) [inline src/core/Core.cpp:755]: thats not what i meant

```
if(eeMousePressed1()) {
	if (getInventoryItemWithLowestDurability("arrow", 1.f) != nullptr || getInventoryItemWithLowestDurability("arrows", 1.f) != nullptr) {
		changeAnimation(io, 1, alist[ANIM_MISSILE_STRIKE_PART_1]);
		io->isHit = false;
	}
}
```

is what i meant

File: src/core/Core.cpp (hunk 4 above)
Comment (fredlllll) [reply src/core/Core.cpp:867]: you forgot to fix that one to

`if (quiver != nullptr)`

**Review by fredlllll — CHANGES_REQUESTED** [review]

**Review by fredlllll — APPROVED** [review]

**Comment by meszaros-lajos-gyorgy** [comment]
merged it into insanity branch manually because it contained conflicts

**Pull request closed without merging**